### DISCUSSION TITLE
Alternate solution to storage container problem

### DISCUSSION THREAD

**Discussion by G-M-twostay** [FakeFishGames/Barotrauma#12241] [Unstable feedback]
### Original Goal
Totally remove the possibility of abusing container items such as _storage containers_ to store a huge amount of items in a single cabinet(I had a short discussion in discord channel with regalis and confirmed that this was the intention). Then, address the heavily reduced storage capabilities of cabinets after the removal of the widely used storage containers.

### Current Solution

1. Remove _storage containers_.
2. Increase stack size of some items in cabinets and crates.
3. Slightly increased the amount of slots in cabinets.

**Benefits**

1. The possibility of abusing _storage containers_ is completely gone.
2. The increased stack size and increased slots make us able to store the same amount of items(or more) using crates and cabinets, given we're talking about items that have increased stack size. For example, I use storage containers to store minerals. Previously, I can store 12*8 minerals in one slot using storage contaienrs. Now, I can still put 32 of them in one slot, and I can utilize crates to store them too. Combined with increased slot amounts, these items only takes less than 3 times the amount of space without storage containers.
3. Players are more motivated to use crates, which, previously, aren't quite useful due to the existent of _storage containers_.

**Downsides**

1. Even though abusing storage containers is no longer possible, it's still possible to abuse _toolbelts_ and _cargo scooters_.
2. Cabinets suffer a dramatically loss of capabilities to store items such as refined gene materials, tools, and weapons. Basically, items that's hard to stack or can't be stack takes 6 times the space as before without storage containers.

### Alternate Apporach
**Disclaimer**
This solution is to address the reduced the storage capabilities of cabinets for non-stackable items, as well as addressing the potential of abusing toolbelts and cargo scooters. This solution is unrelated to whether item stack size are increased, nor if storage containers are removed. 

**Solution**

1. Don't allow putting storage containers, cargo scooters, and toolbelts in normal cabinet slots.
2. Instead, add special slots dedicated for storing these items to the cabinet.
3. (Just an idea). It can be made that large items(2 handed) items can't be stored in normal slots, and that these items, together with container items, can only be stored in these dedicated slots. I haven't really think thoroughly about how this will work though.

**Benefites**

1. No more abusing of any kind of container items in cabinets to store a crazy amount of items in a single cabinet. This number can now be limited and controlled.
2. Indepedent of other changes such as increasing stack size and removing storage containers.
3. Consistency. It kind of felt unreasonable that a single slot in the cabinet can only hold one diving knife while still being able to hold a big cargo scooter that can store 11 diving knives. 
4. Increased storage capabilities for unstackable items like tools, as you can still put them in container items. It's just that this increased capability is a controlled and reasonable amount.

**Downsides**

1. More work for devs by adding tweaking the gui. Basically, cabinets will now look like the inventory of cargo scooters where there are some didcated slots(like cargo scooter's battery slot). However, this isn't really an issue for me because I'm not a dev :).
2. Compatibility issue. There is a need to come up with a way to deal with existing container items in cabinets. Here I propose one way. Make it so that if you take these storage containers out of the normal slots you can't put them back in again, instead of just forcing them out and create a mess on the floor.

**Comment by Hazmat616** [comment]
You could just make it you can’t store items holding items

**Reply by aeondez** [reply]
This.

Don't allow items that can store other items to be stored inside any storage container/cabinet/crate/crate shelf.

**Comment by aeondez** [comment]
What about a "large" weapon rack that holds more than one weapon, which would both look cool and hold some of those non-stackable weapons in their own dedicated container to prevent tossing weapons on the floor because cabinet space is at a premium.

Bonus points if we can make accessing it role-locked like secure doors/toxin cabinets/etc.

**Reply by G-M-twostay** [reply]
well, i was going to add a feature in my proposal so that the containers display the large items in the dedicated slots. then i thought this might be to complicated for the devs to do lol.

**Comment by Ek-Videogames** [comment]
This isn't about "exploits" or whatever backwards idea of 'game too easy' you happen to have, the devs say it is allegedly for "performance reasons".

A possible solution is nerf/rollback crafting recipe costs & resource scarcity.
( Shortly after item stacking was first added to the game, the devs started increasing the costs of many recipes in the game, this had the indirect but not unsurprising effect of making the number of resources you have to carry on the submarine much greater. )
If the game engine suddenly can't handle items anymore, then maybe they shouldn't have made it so people have to handle massive amounts of items?

Deleting storage containers from the game won't actually solve any of the reasons why **people are resorting to using them in the first place** and will just make dealing with those problems more annoying. IE; 
- High consumption of resources, meaning players need to hold onto more items for supplies.
Crafting items and other standard constantly consumed items like ammo & fuel rods.

- Pitiful vanilla (or some modded) submarine storage capability, and suboptimal fabricator linking.
There are many types of items in the game, and even with 32x stack limits there's still more types of items that won't stack with each-other. Also they are quite useful as a smaller more-handy portable storage alternative to metal crates, and can be put away in a container when not in use unlike crates.

- Fabricator linking also has it's own major performance flaws, and those issues won't go away when they get rid of storage containers. Fabricator will still have lag issues if its linked to an inventory with a ton of items in it. 
_Maybe the performance improvements they said they are making to item stacks will fix this? Though given every update the last 4 years has made the game run gradually worse I'm cautiously optimistic._

**Reply by Ek-Videogames** [reply]
Hell, if they want to cut out the middleman entirely and just add some kind of absolutely **massive** storage container that links to the fabricator & can somehow be accessed from linked 'access ports' around the ship in different places - I wouldn't entirely be opposed to that.

**Reply by 0n1ime** [reply]
The problems started when the ability to stack items was added. There shouldn't be any items in the game at all that can be stacked. As for consumable items (Chemicals. Materials), it is worth adopting the mechanics of the condition (ammo boxes, books, magazines). For example, in the current version, two irons are needed to make a screwdriver, in the method that I described, one iron is needed, for crafting a screwdriver, it will take 10% of the iron's condition. This means that 10 screwdrivers can be crafted from one iron.

**Reply by Ek-Videogames** [reply]
Yeah, we *HAD* single items (however we also had storage containers, because obviously you're not going to fit more than 10 materials into a locker otherwise) but then people complained about "partial-items" IE that 10% steel bar, so to try to fix the 'granularity' of crafting recipes they made all of the recipes super expensive (also deconstruction got removed for some reason 🤷‍♂️), but then that meant players needed to "hoard" (have enough to actually utilize said recipes) resources more and this also impacted the economy. Problem is removing storage containers didn't make the economy balanced, all it did is change how inventory micromanagement is going to be into a more annoying and time consuming one.

**Comment by bookslayer10** [comment]
Removing storage containers makes it easier to store low and medium ammounts of resources, which is nice, but it does make storing lots of resources harder. Currently this is just half of a solution, and debatable whether or not it makes the game better.

There should be an improvement to storing lots of resources to make it easier. I'd be fine with it being a submarine upgrade that the crew needs to pay for, because then a powerful system would need to be earned.

One solution would be a central cargo hold ME style system, with submarine upgrades to improve storage space and item access time. Another option is making crates easier to use, by letting crate racks link to fabricators. Add a sort feature to crates, which organizes the resources in all the crates in a rack into individual crates, with one for the leftovers. Simple mass storage, one crate per resource type. Maybe the racks meant for crates wouldn't count towards cargo missions, and plain crates could be constructed for free through a mechanic/assistant skill.

**Comment by Virtuosopainting24** [comment]
Tip for washing plastic bags: turn it inside out, stick you hand inside, add a dot of detergent and "wash your hands".  Works a treat!
